Scripts/Runtime/Entities/Lifecycle/Spawning/EntitySpawnSystem.cs

```diff
@@ -0,0 +1,424 @@
+        /// <returns>The created <see cref="Entity"/></returns>
+        public Entity SpawnImmediate<TEntitySpawnDefinition>(Entity prototype, TEntitySpawnDefinition spawnDefinition, bool shouldDestroyPrototype = false)
+            where TEntitySpawnDefinition : unmanaged, IEntitySpawnDefinition
+        {
+            EntityPrototypeSpawner<TEntitySpawnDefinition> entitySpawner = GetOrCreateEntitySpawner<EntityPrototypeSpawner<TEntitySpawnDefinition>, TEntitySpawnDefinition>();
+            return entitySpawner.SpawnImmediate(prototype, spawnDefinition, shouldDestroyPrototype);
+        }
+
+        //TODO: Implement a SpawnImmediate that takes in a NativeArray or ICollection if needed.
+
+        private void EnableSystem(IEntitySpawner entitySpawner)
+        {
+            //By using this, we're writing immediately, but will need to execute later on when the system runs
+            Enabled = true;
+            m_ActiveEntitySpawners.Add(entitySpawner);
+        }
+
+        protected override void OnUpdate()
+        {
+            Dependency = ScheduleActiveEntitySpawners(Dependency);
+            
+            //Ensure we're turned back off
+            m_ActiveEntitySpawners.Clear();
+            Enabled = false;
+        }
+
+        private JobHandle ScheduleActiveEntitySpawners(JobHandle dependsOn)
+        {
+            NativeArray<JobHandle> dependencies = new NativeArray<JobHandle>(m_ActiveEntitySpawners.Count + 1, Allocator.Temp, NativeArrayOptions.UninitializedMemory);
+            dependencies[^1] = m_EntityArchetypes.AcquireAsync(AccessType.SharedRead, out NativeParallelHashMap<long, EntityArchetype> entityArchetypesLookup);
+
+            //All the active spawners that need to go this frame are given a chance to go ahead and run their spawn jobs
+            int index = 0;
+            foreach (IEntitySpawner entitySpawner in m_ActiveEntitySpawners)
+            {
+                EntityCommandBuffer ecb = m_CommandBufferSystem.CreateCommandBuffer();
+                dependencies[index] = entitySpawner.Schedule(dependsOn, ref ecb, entityArchetypesLookup);
+                index++;
+            }
+
+            dependsOn = JobHandle.CombineDependencies(dependencies);
+            m_EntityArchetypes.ReleaseAsync(dependsOn);
+            m_CommandBufferSystem.AddJobHandleForProducer(dependsOn);
+            return dependsOn;
+        }
+
+        private TEntitySpawner GetOrCreateEntitySpawner<TEntitySpawner, TEntitySpawnDefinition>()
+            where TEntitySpawner : IEntitySpawner, new()
+            where TEntitySpawnDefinition : unmanaged, IEntitySpawnDefinition
+        {
+            Type spawnerType = typeof(TEntitySpawner);
+            Type definitionType = typeof(TEntitySpawnDefinition);
+            // ReSharper disable once InvertIf
+            if (!m_EntitySpawners.TryGetValue(spawnerType, out IEntitySpawner entitySpawner))
+            {
+                // ReSharper disable once SuggestVarOrType_SimpleTypes
```

Comment:
Sure, I'd be interested to see what its like. 

Personally I like them in my face because then it's easy to see when i'm off-style and it's a quick fix. If i'm doing it intentionally, the explicit warning disable lets me know I don't agree with this, but i'm doing it anyway